File: src/docStats.ts
export interface DocStats {
  chars: number;
  words: number;
}

export function computeDocStats(text: string): DocStats {
  const trimmed = text.trim();
  return {
    chars: text.length,
    words: trimmed === "" ? 0 : trimmed.split(/\s+/).length,
  };
}

File: src-tauri/notes-iced/src/logic/doc_stats.rs
//! Port of `src/docStats.ts`. TODO(unit 1): port behaviour and tests.

#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
pub struct DocStats {
    pub chars: usize,
    pub words: usize,
}

pub fn compute(text: &str) -> DocStats {
    DocStats {
        chars: text.chars().count(),
        words: text.split_whitespace().count(),
    }
}
